TeamCode/src/main/java/org/firstinspires/ftc/teamcode/Components/DriveTrain.java

```diff
@@ -0,0 +1,119 @@
+package org.firstinspires.ftc.teamcode.Components;
+import com.qualcomm.robotcore.eventloop.opmode.OpMode;
+import com.qualcomm.robotcore.hardware.DcMotor;
+
+/**
+ * Drivetrain Controls:
+ * Gamepad: 1
+ * Joystick: Left
+ */
+public class DriveTrain{
+
+    OpMode opMode;
+
+    private boolean driving = false;
+    private boolean turning = false;
+    private final double TICKS_PER_INCH = 1000/14;
+    private  final double circumference = 14 * Math.sqrt(2) * Math.PI;
+    private  final double degree = circumference / 360;
+
+    private final DcMotor[][] motors = new DcMotor[2][2];
+
+    public DriveTrain(OpMode opMode) {
+        this.opMode = opMode;
+    }
+
+    public void init(boolean debug){
+        motors[0][0] = opMode.hardwareMap.dcMotor.get("FrontLeft");
+        motors[0][1] = opMode.hardwareMap.dcMotor.get("BackLeft");
+        motors[1][0] = opMode.hardwareMap.dcMotor.get("FrontRight");
+        motors[1][1] = opMode.hardwareMap.dcMotor.get("BackRight");
+
+        if(debug){
+            opMode.telemetry.addData("INFO:","Drivetrain Init");
+        }
+    }
+
+    public void run(boolean debug) {
+        double leftPower;
+        double rightPower;
+
+        leftPower = opMode.gamepad1.left_stick_y - opMode.gamepad1.left_stick_x;
+        rightPower = opMode.gamepad1.left_stick_y + opMode.gamepad1.left_stick_x;
+
+        motors[0][0].setPower(leftPower);
+        motors[0][1].setPower(leftPower);
+        motors[1][0].setPower(rightPower);
+        motors[1][1].setPower(rightPower);
+
+        if(debug){
+            opMode.telemetry.addData("Left Power", leftPower);
+            opMode.telemetry.addData("Right Power", rightPower);
+        }
+
+    }
+
+    // AUTO DRIVING
+    private int distanceTraveled = 0;
+    private int driveStartingPos = 0;
+    public boolean autoDrive(int distance){
+        distance = (int) (distance * TICKS_PER_INCH);
+        if (!(driving)) {
+            for (int i = 0; i < motors[0].length; i++) {
+                driveStartingPos = motors[0][i].getCurrentPosition();
```

Comment:
I would reset the encoders, the way you are doing it could cause problems if not all the motors start at the same position because you are only checking the encoder of the motor at [0][0], yet driveStartingPos will be set to the initial value of the encoder of the motor at [1][1]. It also makes the math a bit easier